packages/shopify-api/lib/base-types.ts

```diff
@@ -5,32 +5,105 @@ import {AuthScopes} from './auth/scopes';
 import {BillingConfig} from './billing/types';
 import {ApiVersion, LogSeverity} from './types';
 
+/**
+ * A function used by the library to log events related to Shopify.
+ */
 export type LogFunction = (severity: LogSeverity, msg: string) => void;
 
 export interface ConfigParams<
   Resources extends ShopifyRestResources = ShopifyRestResources,
   Future extends FutureFlagOptions = FutureFlagOptions,
 > {
+  /**
+   * The API key for your app.
+   *
+   * You can find this in the Partner Dashboard.
```

Comment:
In the past folks have gotten confused because in the partner dashboard it is called client ID

Maybe 
```suggestion
   * Also known as Client ID in your Partner Dashboard.
```